exp/api/v1alpha4/azuremanagedmachinepool_webhook.go

```diff
@@ -114,6 +118,25 @@ func (r *AzureManagedMachinePool) ValidateDelete(client client.Client) error {
 	return errors.Wrapf(r.validateLastSystemNodePool(client), "if the delete is triggered via owner MachinePool please refer to trouble shooting section in https://capz.sigs.k8s.io/topics/managedcluster.html")
 }
 
+// Validate the AzureManagedMachinePool and return an aggregate error.
+func (r *AzureManagedMachinePool) Validate(client client.Client) error {
```

Comment:
Is this needed now? Can we instead add a test for default webhook?

Reply:
Thanks for catching that. Removed the validation logic. Added tests for default webhook.